src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```
@@ -233,4 +234,24 @@ public function findCollectionBreadcrumbById($id)
             return [];
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findCollectionByKey($key)
+    {
+        $queryBuilder = $this->createQueryBuilder('collection')
+            ->leftJoin('collection.meta', 'collectionMeta')
+            ->leftJoin('collection.defaultMeta', 'defaultMeta')
+            ->where('collection.key = :key');
+
+        $query = $queryBuilder->getQuery();
+        $query->setParameter('key', $key);
+
+        try {
+            return $query->getSingleResult();
+        } catch (UnexpectedResultException $ex) {
```

Do you really want to silently return null for the `NonUniqueResultException`? That doesn't feel right to me, because that would be a serious violation of database constraints, for the `NoResultException` that seems right.